src/components/Communicator/Communicator.actions.js

```diff
@@ -291,3 +300,34 @@ export function updateDefaultBoardsIncluded(boardAlreadyIncludedData) {
     defaultBoardsIncluded: boardAlreadyIncludedData
   };
 }
+
+export function concatDefaultBoardIdToBlacklist(boardId) {
+  const getActiveCommunicator = getState => {
+    return getState().communicator.communicators.find(
+      c => c.id === getState().communicator.activeCommunicatorId
+    );
+  };
+  return (dispatch, getState) => {
+    if (!ALL_DEFAULT_BOARDS.map(({ id }) => id).includes(boardId)) return;
+    const updatedCommunicatorData = { ...getActiveCommunicator(getState) };
+
+    const concatBoardIdIfNecessary = () => {
+      if (!updatedCommunicatorData?.defaultBoardBlackList.includes(boardId))
+        return updatedCommunicatorData?.defaultBoardBlackList.concat(boardId);
+      return updatedCommunicatorData?.defaultBoardBlackList;
+    };
+
+    updatedCommunicatorData.defaultBoardBlackList = updatedCommunicatorData?.defaultBoardBlackList
+      ? concatBoardIdIfNecessary()
+      : [boardId];
+
+    dispatch(verifyAndUpsertCommunicator(updatedCommunicatorData));
+    return dispatch(upsertApiCommunicator(updatedCommunicatorData))
```

Comment:
This async action dispatch can be avoided such as the communicator gets updated later on. Just update the local state and be sure to get the updated state before the call of the `upsertApiCommunicator` action